### PR TITLE
Jest stimulus unit tests

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Partial rework of https://github.com/pypa/warehouse/pull/3913, mainly using [Jest](https://jestjs.io/) as a testing framework with [jest-dom](https://github.com/testing-library/jest-dom) helpers for assertions.

~Includes a single test suite for the password toggle controller.~ Includes tests for all Stimulus controllers.

Update Babel to v7 in the process.

### CODE REVIEW

**Comment by yeraydiazdiaz** [comment]
Probably best to leave this open until we have a complete test suite, but I'd appreciate any feedback.

Personally I think it's less invasive and straightforward than the previous implementation.

**Comment by yeraydiazdiaz** [comment]
This is now ready for review, cc @di @nlhkabu @woodruffw

**Comment by yeraydiazdiaz** [comment]
Yeah, definitely, I'll add that.

**Comment by di** [comment]
@yeraydiazdiaz Looks like you have some linting errors now.

**Comment by yeraydiazdiaz** [comment]
Yep, pushed to soon, it's all fixed now 👍

**Comment by di** [comment]
@nlhkabu, can you try this out a bit locally and confirm that the upgraded dependencies don't cause any problems?

**Comment by yeraydiazdiaz** [comment]
Could we also double check the piece of actual logic change in the notifications controller, please?

**Comment by nlhkabu** [comment]
No problem running this locally. :+1:

File: docs/development/frontend.rst [hunk 1]
```diff
@@ -17,6 +17,37 @@ however, you can trigger a manual build of them by installing all of the
 dependencies using ``npm install`` and then running ``gulp dist``.
 
 
+Tests
+-----
+
+The JavaScript codebase includes tests that can be ran via
+``make static_tests`` which is also ran with the ``make tests`` target.
+Both targets will run the tests in the Docker enviroment but they
+may also be ran locally using ``npm run test``.
```
Comment (nlhkabu) [inline docs/development/frontend.rst]: It could be useful here to note the version of node/npm required to run the tests locally.

Comment (yeraydiazdiaz) [reply docs/development/frontend.rst]: Good point, I don't think there's any mention to installing nodejs in the docs, I'll add that. 👍

Comment (yeraydiazdiaz) [reply docs/development/frontend.rst]: This has been addressed.

**Pull request merged**